vividus/src/main/java/org/vividus/bdd/steps/BddVariableSteps.java

```diff
@@ -177,18 +180,41 @@ public boolean compareVariables(Object variable1, ComparisonRule condition, Obje
      * @param table ExamplesTable with parameters to compare with
      */
     @Then("`$variable` is equal to table ignoring extra columns:$table")
+    @SuppressWarnings("unchecked")
     public void tablesAreEqualIgnoringExtraColumns(Object variable, ExamplesTable table)
     {
-        if (!(variable instanceof Map))
-        {
-            throw new IllegalArgumentException("'variable' should be instance of map");
-        }
-        Map<String, String> expectedMap = MapUtils.convertSingleRowExamplesTableToMap(table);
-        List<EntryComparisonResult> results = ComparisonUtils.checkMapContainsSubMap((Map<?, ?>) variable, expectedMap);
+        isTrue(variable instanceof Map, "'variable' should be instance of map");
+        Map<Object, Object> actualMap = expandListValuesWithSingleElement((Map<Object, Object>) variable);
+
+        List<Parameters> rows = table.getRowsAsParameters(true);
+        isTrue(rows.size() == 1, "ExamplesTable should contain single row with values");
+
+        Map<String, String> expectedMap =  rows.get(0).values();
+        List<EntryComparisonResult> results = ComparisonUtils.checkMapContainsSubMap(actualMap, expectedMap);
         publishMapComparisonResults(List.of(results));
         softAssert.assertTrue(TABLES_ARE_EQUAL, results.stream().allMatch(EntryComparisonResult::isPassed));
     }
 
+    private Map<Object, Object> expandListValuesWithSingleElement(Map<Object, Object> map)
```

Comment:
if we have 10 entries in the map where first 9 entries have a value List with size 1, but the last one is different, we will eventually return the map passed as param and put operation on expandedMap will be redundant, so does it make sense to check first that the input map has list with size 1? (for example via allMatch)

Reply:
in this way it'll be required to traverse the map twice and execute similar logic, if you'd like, you can create a benchmark

Reply:
even if we imagine we have a dataset where the double for is slower it's a bit weird to populate map that is potentially will never be used

Reply:
nothing weird: we're trying to construct a new object, if it doesn't match the rule, we just throw it away
for simple objects: it could be easy to perform the full check and then construct the object, for complex objects: sometimes the check and the iteration could be of higher cost, than creation of object itself, so some flexibility is required